src/librustc_trans/trans/consts.rs

```diff
@@ -664,12 +662,19 @@ fn const_expr_unadjusted<'a, 'tcx>(cx: &CrateContext<'a, 'tcx>,
             adt::const_get_field(cx, &*brepr, bv, vinfo.discr, idx.node)
         },
         hir::ExprIndex(ref base, ref index) => {
+            //honor the const_indexing feature if this is in a CONST expression
+            if !cx.sess().features.borrow().const_indexing && trueconst == TrueConst::Yes {
```

Comment:
I had to add this check in order to get the `compile-fail` test I added to pass. Maybe this is the wrong place to add the check though?